concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/BakerPoolStatus.java

```diff
@@ -60,6 +61,22 @@ public class BakerPoolStatus {
      */
     private final CCDAmount allPoolTotalCapital;
 
+    public Optional<CCDAmount> getBakerEquityCapital() {
+        return Optional.ofNullable(bakerEquityCapital);
+    }
+
+    public Optional<CCDAmount> getDelegatedCapital() {
+        return Optional.ofNullable(delegatedCapital);
+    }
+
+    public Optional<CCDAmount> getDelegatedCapitalCap() {
+        return Optional.ofNullable(delegatedCapitalCap);
+    }
+
+    public Optional<BakerPoolInfo> getPoolInfo() {
+        return Optional.ofNullable(poolInfo);
+    }
+
```

Comment:
I don't know a lot of java, but how would you even access these before? Is there something implicit getter implementation going on with one of the annotations `@Data` for example? and does the implementation of `public Optional<X> getY()` override the implementation otherwise implicitly provided?

Reply:
There is a plugin which generates getters, but for fields which sometimes are expected to be `null` there are explicit getters returning `Optional`, they are added manually. It is not my idea, I followed the code written last year by Emil Lai and Prav Sharma.

Reply:
Yeah I read a little about `@Data` and `@Value` annotations from `lombok` yesterday 😄

Changing a function signature from `CCDAmount getBakerEquityCapital()` to `Optional<CCDAmount> getBakerEquityCapital()` _is_ a breaking change in my mind though.. can you explain why you think it is not?

Reply:
Of course, you are right! I explained how the getters work but haven't seen a breaking change introduced by them 😅